src/object/player.cpp

```diff
@@ -324,10 +354,43 @@ Player::update(float dt_sec)
 
   // extend/shrink tux collision rectangle so that we fall through/walk over 1
   // tile holes
-  if (fabsf(m_physic.get_velocity_x()) > MAX_WALK_XM) {
-    m_col.set_width(RUNNING_TUX_WIDTH);
-  } else {
-    m_col.set_width(TUX_WIDTH);
+
+  Rectf lookahead = get_bbox();
+  lookahead.set_right(lookahead.get_left() + 62);
+  bool pathBlocked = !Sector::get().is_free_of_statics(lookahead);
+  
+  if((m_swimming || m_water_jump) && is_big() && !pathBlocked)
```

Comment:
Maybe use pathFree instead of pathBlocked and avoid double boolean negation (-> improve readability)